### PR TITLE
PLFM-7490: Enable RDS monitoring

### PR DESCRIPTION
**Pull request by xschildw**
This PR enables RDS enhanced monitoring and insights on prod RDS instances.

### CODE REVIEW

File: src/main/resources/templates/repo/main-repo-shared-resources-template.json.vpt [hunk 1]
```diff
@@ -119,6 +140,14 @@
 					}
 				],
 				"StorageEncrypted": true
+                #if((${stack} == 'prod') && (${EnableRdsEnhancedMonitoring} == "true"))
```
Comment (marcomarasca) [inline src/main/resources/templates/repo/main-repo-shared-resources-template.json.vpt]: Maybe we can ignore the "prod" switch here and put this property in the "repo-defaults.properties" that is used to build each DB descriptor. In this way we can set it to false by default and enabled it only in the prod deployment.

**Review by marcomarasca — CHANGES_REQUESTED** [review]

File: src/main/resources/templates/repo/main-repo-shared-resources-template.json.vpt [hunk 2]
```diff
@@ -119,6 +140,14 @@
...
+                ,
+                "EnablePerformanceInsights": true,
+                "MonitoringInterval": "60",
```
Comment (marcomarasca) [inline src/main/resources/templates/repo/main-repo-shared-resources-template.json.vpt]: Maybe we can start with 15 or 30

**Review by marcomarasca — CHANGES_REQUESTED** [review]
See comments

**Review by marcomarasca — APPROVED** [review]

File: src/main/resources/templates/repo/main-repo-shared-resources-template.json.vpt [hunk 3]
```diff
@@ -118,7 +139,15 @@
 						"Ref": "${stack}${instance}VpcDBSecurityGroup"
 					}
 				],
-				"StorageEncrypted": true
+				"StorageEncrypted": true,
+                "EnablePerformanceInsights": ${EnableRdsEnhancedMonitoring}
```
Comment (xschildw) [inline src/main/resources/templates/repo/main-repo-shared-resources-template.json.vpt:143]: Moved this out of condition

File: src/test/java/org/sagebionetworks/template/repo/RepositoryTemplateBuilderImplTest.java [hunk 4]
```diff
@@ -368,6 +437,7 @@ public void testBuildAndDeployDev() throws InterruptedException {
 		JSONObject dbProps = (JSONObject)repoDB.get("Properties");
 		assertFalse(dbProps.has("DBSnapshotIdentifier"));
 		assertTrue(dbProps.has("DBName"));
+		validateResouceDatabaseInstance(resources, stack, "true");
```
Comment (xschildw) [inline src/test/java/org/sagebionetworks/template/repo/RepositoryTemplateBuilderImplTest.java:440]: Added missing check

File: src/test/java/org/sagebionetworks/template/repo/RepositoryTemplateBuilderImplTest.java [hunk 5]
```diff
@@ -492,6 +565,19 @@ public void validateResouceTablesDatabase(JSONObject resources, String stack) {
 		assertEquals(Boolean.FALSE, properties.get("MultiAZ"));
 		assertEquals(stack+"-101-table-1", properties.get("DBInstanceIdentifier"));
 		assertEquals(stack+"101", properties.get("DBName"));
+		validateEnhancedMonitoring(properties, enableEnhancedMonitoring);
+	}
+
+	public void validateEnhancedMonitoring(JSONObject props, String enableEnhancedMonitoring) {
+		assertTrue(props.has("EnablePerformanceInsights"));
+		assertEquals(Boolean.parseBoolean(enableEnhancedMonitoring), props.get("EnablePerformanceInsights"));
```
Comment (xschildw) [inline src/test/java/org/sagebionetworks/template/repo/RepositoryTemplateBuilderImplTest.java:573]: Moved out of condition

**Review by marcomarasca — APPROVED** [review]

**Pull request merged**